gui.py

```diff
@@ -66,59 +74,122 @@ def run_sss(self):
+                            instrument=self.my_data.extraction_type,
                             reverse=self.reverseCheckBox.isChecked(),
-                            quality=self.qualityComboBox.currentText().lower(),
+                            quality='to_delete',
                             max_iter=self.maxIterationsSpinBox.value()))
         _, self.my_data.sr = sf.read(self.input_location)
         
         if self.evaluation_reference:
            self.my_data.evaluation_results = evaluate(result_wave=self.my_data.audio_wave,
                                              eval_params=EvalParams(self.evaluation_reference))
-        
+           self.evaluation_reference = ''
+
         self.save_ui.outputNameLabel.setText(f'{self.my_data.extraction_type}')
         widget.setCurrentIndex(2)
 
 
+class GUILoading(Ui_LoadingWindow):
+    def __init__(self):
+        self.window = None
+        self.loading_gif = QMovie('resources\loading.gif')
+
+    def add_features(self, window):
+        self.window = window
+
+        self.gifLabel.setMovie(self.loading_gif)
+        self.loading_gif.start()
+
+
 class GUISave(Ui_SaveWindow):
     def __init__(self, my_data):
-        self.my_data = my_data
-
         self.window = None
+        self.my_data = my_data
+        self.player = QMediaPlayer()
 
     def add_features(self, window):
         self.window = window
 
         self.saveEvaluationButton.setDisabled(True)
+        self.runButton.setDisabled(True)
 
         self.playButton.clicked.connect(self.play_handler)
+        self.runButton.clicked.connect(self.run_handler)
+        self.volumeDownButton.clicked.connect(self.volumeDown_handler)
+        self.volumeUpButton.clicked.connect(self.volumeUp_handler)
         self.saveOutputButton.clicked.connect(self.save_output_handler)
         self.saveEvaluationButton.clicked.connect(self.save_evaluation_handler)
         self.returnButton.clicked.connect(self.return_handler)
 
     def save_output_handler(self):
         output_location = QtWidgets.QFileDialog.getExistingDirectory(self.window,
                                                                           "Choose separation output directory", "c:\\")
-        save(result_wave=self.my_data.audio_wave,
-             method=self.my_data.method,
-             save_params=SaveWavParams(
-                 output_path=output_location + '/extracted',
-                 extraction_type=self.my_data.extraction_type,
-                 sample_rate=self.my_data.sr))
+        if output_location:
+            save(result_wave=self.my_data.audio_wave,
+                 method=self.my_data.method,
+                 save_params=SaveWavParams(
+                     output_path=output_location,
+                     instrument=self.my_data.extraction_type,
+                     sample_rate=self.my_data.sr,
+                     input_track=self.my_data.input_track_name))
 
     def save_evaluation_handler(self):
```

Comment:
Wciśnięcie "anuluj" podczas wyboru ekstrakcji skutkuje odpaleniem programu z ` ` jako nazwą pliku referencyjnego. Należy to zmienić, bo powoduje to `RuntimeError` i crashuje appkę.